base.go

```diff
@@ -0,0 +1,75 @@
+package rpc
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+type base struct {
+	node        string
+	idGenerator func(string) interface{}

```

Comment:
How about using more concrete type for the return type of `idGenerator`?

Reply:
I replaced the type of `idGenerator` to `func(string) string`